### DISCUSSION TITLE
dfx generate is not compiling the candid if the canister is not deployed

### DISCUSSION THREAD

**Discussion by ozwaldorf** [dfinity/sdk#2733] [Feature Requests]
Currently, if you have a child canister that is in the project, but does not get deployed directly by dfx and rather a parent canister, you cannot generate the bindings. The generation only works after creating the canister id. A usual workaround is specifying the management or anonymous principal id in canister_ids.json, but as generate only reads the local ids and they are always ephemeral, it requires manual intervention any time dfx is reset.

I suggest removing the canister id requirements for dfx generate, and only rely on the candid file existing

**Comment by viviveevee** [comment]
Part of [SDK-693](https://dfinity.atlassian.net/browse/SDK-693)

**Comment by ByronBecker** [comment]
@sesi200 Can this be added as a feature request? (To discussions)

**Reply by viviveevee** [reply]
Of course! I thought it wasn't necessary to convert but now that I didn't get around to do it yet it makes sense to enable the upvoting on the issue

**Comment by paulyoung** [comment]
Can you use `didc bind hello.did -t js` as a workaround in the meantime?

**Comment by viviveevee** [comment]
This will be partially resolved in #2988

**Comment by viviveevee** [comment]
This will be fully resolved in #3093
